test/jdk/java/net/URLConnection/UNCTest.java

```diff
@@ -28,44 +28,42 @@
  * @modules java.base/sun.net.www.protocol.file
  * @library /test/lib
  * @summary  Check that URL.openConnection() doesn't open connection to UNC
- * @run main UNCTest
  */
 
 import jtreg.SkippedException;
 import sun.net.www.protocol.file.FileURLConnection;
 
 import java.io.File;
 import java.net.InetAddress;
+import java.net.URI;
 import java.net.URL;
 import java.net.URLConnection;
 
 public class UNCTest {
     public static void main(String args[]) throws Exception {
         // Get the "computer name" for this host
         String hostName = InetAddress.getLocalHost().getHostName();
-        // Skip the test if Administrative Shares is disabled
-        skipIfAdministrativeSharesDisabled(hostName);
-        // Should always exist with Administrative Shares enabled
-        URL url = new URL("file://" + hostName +"/C$/Windows");
+
+        // UNC path which always exist with Administrative Shares enabled
+        String path = "\\\\" + hostName + "\\C$\\Windows";
+
+        // Skip test if Administrative Shares is disabled
+        if (! new File(path).exists()) {
+            throw new SkippedException("Administrative shares not enabled");
+        }
+
+        // File URL for the UNC path
+        URL url = new URI("file:" + path.replace('\\', '/')).toURL();
+
```

Comment:
My gut feeling is that using `path.replace` here makes what's going on more obscure. I would keep the original string: `"file://" + hostName +"/C$/Windows"`

Reply:
> My gut feeling is that using `path.replace` here makes what's going on more obscure. I would keep the original string: `"file://" + hostName +"/C$/Windows"`

You're right, String::replace was not very helpful here. Reverted as per your suggestion.